pingcap/tidb
# config, sessionctx: deprecate streaming config

### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/24868 

Problem Summary:
Streaming has been deprecated. But it maybe enable by mistake.
### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:
1. Avoid to enable streaming by config file.
2. Hide "tidb_enable_streaming" user-variable. 

How it Works:

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
Test Config
```
➜  bin git:(deprecate-streaming) cat test.config     
enable-streaming = true
➜  bin git:(deprecate-streaming) ./tidb-server -config=./test.config                             
config file ./test.config contained invalid configuration options: enable-streaming; check TiDB manual to make sure this option has not been deprecated and removed from your TiDB version if the option does not appear to be a typo
[2021/07/14 15:55:04.065 +08:00] [INFO] [printer.go:33] ["Welcome to TiDB."] ["Release Version"=v5.2.0-alpha-197-g2f2068360] [Edition=Community] ["Git Commit Hash"=2f20683600f4a239570f6ddb7daf656af2ead8d1] ["Git Branch"=deprecate-streaming] 
.....

[tidb]> select @@tidb_enable_streaming;
+-------------------------+
| @@tidb_enable_streaming |
+-------------------------+
|                       0 |
+-------------------------+
1 row in set (0.000 sec)
```
Test hidden variables: (I add the test case in unit test)
```
[tidb]> show variables like '%tidb_enable_streaming%';
Empty set (0.003 sec)
```
- No code

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

- No release note
